### ISSUE TITLE
[client-90051] publish error AgoraRTCException: AgoraRTCError INVALID_OPERATION: Can't publish stream, haven't joined yet!

### ISSUE THREAD

**Issue by elenitaex5** [AgoraIO-Community/Web-React-UIKit#28]
Hi everyone!
A Bug is happening to me really suddenly. Everything worked perfectly but, in one of our test this error has to start appearing. 
Token and channel are well formed and created, nothing has change in our back neither in front.

When i start the call I can see my video stream for nearly 0,5 seconds and suddenly turn to black. 
<img width="444" alt="image" src="https://user-images.githubusercontent.com/14013852/213201449-20d98018-d1e7-437a-9ef0-2c1c6d7e9ba4.png">

Does anybody know what is happening??

**Comment by EkaanshArora** [comment]
From a first glance it looks like an error because of the token. 
Are you updating the token? what's the validity of the token set?

**Comment by elenitaex5** [comment]
We generate a new token each different call. It happens with new tokens, but in our other app we still see video stream right. We have two different frontends with same backend. Call is created exactly with same code in both. But before this error we could see two video streams with no problem and now we have this error in one of them only, no code has been changed. 

This is our function to generate tokens 
```js
return RtcTokenBuilder.buildTokenWithUid(
        AGORA_APP_ID,
        AGORA_APP_CERTIFICATE,
        room,
        roomId,
        RtcRole.PUBLISHER,
        currentTime + 5184000 /* + ONE DAY */
      );
```
      
 And tokens are new each call. 
 
 In other test we had expired token error but not this time.
